src/routes/safe/components/Transactions/TxList/TxSingularDetails.tsx

```diff
@@ -97,9 +99,31 @@ const TxSingularDetails = (): ReactElement => {
         },
       ],
     }
-    // And add it to the corresponding list in the store
-    dispatch(isTxQueued(listItemTx.txStatus) ? addQueuedTransactions(payload) : addHistoryTransactions(payload))
-  }, [fetchedTx, chainId, dispatch])
+
+    // Add historical transaction
+    if (!isTxQueued(listItemTx.txStatus)) {
+      dispatch(addHistoryTransactions(payload))
+      return
+    }
+
+    // Don't add queued transaction until queue list has automatically polled
+    if (!transactions || (transactions.next.count === 0 && transactions.queue.count === 0)) {
+      return
+    }
+
+    // Prepend label to queue transaction payload
+    const isNext = transactions.next.transactions.some(([, txs]) => txs.some(({ id }) => id === listItemTx.id))
+    payload.values = [
+      {
+        label: isNext ? 'next' : 'queued',
```

Comment:
Once the automatically polled transactions are populated, we can determine if the payload contains a `next` or `queued` transaction after comparing the `id` to the store.